src/utils.js

```diff
@@ -75,7 +75,22 @@ function getNetworkedEntity(entity) {
 module.exports.getNetworkedEntity = getNetworkedEntity;
 
 module.exports.takeOwnership = function(entity) {
-  getNetworkedEntity(entity).setAttribute("networked", { owner: NAF.clientId });
+  const networkedEntity = getNetworkedEntity(entity);
+  if(!networkedEntity) //bug? sometimes this is null
+  {
+    return false;
+  }
+
+  const owner = networkedEntity.getAttribute("networked").owner;
+  const lastOwnerTime = networkedEntity.getAttribute("networked").lastOwnerTime;
+  const now = this.now();
+  if(owner && owner !== NAF.clientId && lastOwnerTime < now) {
```

Comment:
Can we move the rest of this util function into the networked component?

Maybe:
```
const networkedEntity = getNetworkedEntity(entity);

if(!networkedEntity) {
  return false;
}

return networkedEntity.takeOwnership();
```

This should allow you to move `lastOwnerTime` back to an instance variable on the component without something like `networkedEntity.components.networked.lastOwnerTime`.